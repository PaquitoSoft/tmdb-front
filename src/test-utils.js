import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppProvider } from './components/app-context/app-context';

const userIdMock = 'hskfjahshjdjskla';

const imagesConfigMock = {
	"baseUrl":"https://image.tmdb.org/t/p/",
	"backdropSizes":["w300","w780","w1280","original"],
	"logoSizes":["w45","w92","w154","w185","w300","w500","original"],
	"posterSizes":["w92","w154","w185","w342","w500","w780","original"],
	"profileSizes":["w45","w185","h632","original"],
	"stillSizes":["w92","w185","w300","original"]
};

const renderers = {
	'shallow': shallow,
	'mount': mount
};

function AppContextMock({ children, apiClientMock }) {
	return (
		<AppProvider 
			apiClient={apiClientMock} 
			userId={userIdMock} 
			imagesConfig={imagesConfigMock}
		>
			{children}
		</AppProvider>
	);
}

export function render(
	component, 
	options = { renderType: 'shallow' }
) {
	return renderers[options.renderType](component);
}

export function validateSnapshot(
	component, 
	options = { renderType: 'shallow' }
) {
	const wrapper = render(component, options);
	expect(wrapper).toMatchSnapshot();
	return wrapper;
}

export function renderInAppContext(
	component, 
	options = {}
) {
	const _options = {
		renderType: 'mount',
		apiClientQueryMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		apiClientMutationMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		...options
	};

	const apiClientMock = {
		sendQuery: _options.apiClientQueryMock,
		sendMutation: _options.apiClientMutationMock
	};
	
	const wrapper = render(
		(
			<AppContextMock apiClientMock={apiClientMock}>
				{component}
			</AppContextMock>
		),
		_options
	);

	return wrapper.find(AppProvider).childAt(0);
}

export const clone = object => JSON.parse(JSON.stringify(object));
