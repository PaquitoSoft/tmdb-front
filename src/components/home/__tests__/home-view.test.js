import React from 'react';
import { validateSnapshot } from '../../../test-utils';

import HomeView from '../home-view';

import topRatedServerData from './top_rated-tv-shows-fixture.json';
import popularServerData from './popular-tv-shows-fixture.json';

describe('TvShowDetailView', () => {

	function validateHomeView(urlQueryString, serverData) {
		const _urlQueryString = new URLSearchParams(urlQueryString);
		validateSnapshot(
			<HomeView 
				data={serverData}
				urlQueryString={_urlQueryString}
			/>
		);
	}

	it('Should render Popular tv shows list when URL has no filter', () => {
		validateHomeView('', popularServerData);
	});

	it('Should render Popular tv shows list when URL has that filter', () => {
		validateHomeView('?filter=popular', popularServerData);
	});

	it('Should render Top Rated tv shows list when URL has that filter', () => {
		validateHomeView('?filter=top_rated', topRatedServerData);
	});

});
