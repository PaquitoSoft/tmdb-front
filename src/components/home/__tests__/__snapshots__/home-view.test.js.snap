// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowDetailView Should render Popular tv shows list when URL has no filter 1`] = `
<section
  className="home-view"
>
  <div
    className="home-view__type-selector"
  >
    <FilterLink
      filterCode="popular"
      selectedFilter="popular"
    />
    <FilterLink
      filterCode="top_rated"
      selectedFilter="popular"
    />
  </div>
  <div
    className="home-view__grid"
  >
    <TvShowCard
      key="82856"
      tvShow={
        Object {
          "backdropImagePath": "/o7qi2v4uWQ8bZ1tW3KI0Ztn2epk.jpg",
          "firstAirDate": "2019-11-12",
          "id": 82856,
          "isFavorite": false,
          "name": "The Mandalorian",
          "posterPath": "/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
          "votesAverage": 7.9,
        }
      }
    />
    <TvShowCard
      key="68507"
      tvShow={
        Object {
          "backdropImagePath": "/9yKCJTOh9m3Lol2RY3kw99QPH6x.jpg",
          "firstAirDate": "2019-11-03",
          "id": 68507,
          "isFavorite": false,
          "name": "His Dark Materials",
          "posterPath": "/xOjRNnQw5hqR1EULJ2iHkGwJVA4.jpg",
          "votesAverage": 7.5,
        }
      }
    />
    <TvShowCard
      key="60625"
      tvShow={
        Object {
          "backdropImagePath": "/mzzHr6g1yvZ05Mc7hNj3tUdy2bM.jpg",
          "firstAirDate": "2013-12-02",
          "id": 60625,
          "isFavorite": false,
          "name": "Rick and Morty",
          "posterPath": "/qJdfO3ahgAMf2rcmhoqngjBBZW1.jpg",
          "votesAverage": 8.6,
        }
      }
    />
  </div>
</section>
`;

exports[`TvShowDetailView Should render Popular tv shows list when URL has that filter 1`] = `
<section
  className="home-view"
>
  <div
    className="home-view__type-selector"
  >
    <FilterLink
      filterCode="popular"
      selectedFilter="popular"
    />
    <FilterLink
      filterCode="top_rated"
      selectedFilter="popular"
    />
  </div>
  <div
    className="home-view__grid"
  >
    <TvShowCard
      key="82856"
      tvShow={
        Object {
          "backdropImagePath": "/o7qi2v4uWQ8bZ1tW3KI0Ztn2epk.jpg",
          "firstAirDate": "2019-11-12",
          "id": 82856,
          "isFavorite": false,
          "name": "The Mandalorian",
          "posterPath": "/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
          "votesAverage": 7.9,
        }
      }
    />
    <TvShowCard
      key="68507"
      tvShow={
        Object {
          "backdropImagePath": "/9yKCJTOh9m3Lol2RY3kw99QPH6x.jpg",
          "firstAirDate": "2019-11-03",
          "id": 68507,
          "isFavorite": false,
          "name": "His Dark Materials",
          "posterPath": "/xOjRNnQw5hqR1EULJ2iHkGwJVA4.jpg",
          "votesAverage": 7.5,
        }
      }
    />
    <TvShowCard
      key="60625"
      tvShow={
        Object {
          "backdropImagePath": "/mzzHr6g1yvZ05Mc7hNj3tUdy2bM.jpg",
          "firstAirDate": "2013-12-02",
          "id": 60625,
          "isFavorite": false,
          "name": "Rick and Morty",
          "posterPath": "/qJdfO3ahgAMf2rcmhoqngjBBZW1.jpg",
          "votesAverage": 8.6,
        }
      }
    />
  </div>
</section>
`;

exports[`TvShowDetailView Should render Top Rated tv shows list when URL has that filter 1`] = `
<section
  className="home-view"
>
  <div
    className="home-view__type-selector"
  >
    <FilterLink
      filterCode="popular"
      selectedFilter="top_rated"
    />
    <FilterLink
      filterCode="top_rated"
      selectedFilter="top_rated"
    />
  </div>
  <div
    className="home-view__grid"
  >
    <TvShowCard
      key="100"
      tvShow={
        Object {
          "backdropImagePath": "/ok1YiumqOCYzUmuTktnupOQOvV5.jpg",
          "firstAirDate": "2004-05-10",
          "id": 100,
          "isFavorite": false,
          "name": "I Am Not an Animal",
          "posterPath": "/nMhv6jG5dtLdW7rgguYWvpbk0YN.jpg",
          "votesAverage": 9.5,
        }
      }
    />
    <TvShowCard
      key="87108"
      tvShow={
        Object {
          "backdropImagePath": "/cDpR6URVghK5LC7G7NgPkCP8x6E.jpg",
          "firstAirDate": "2019-05-06",
          "id": 87108,
          "isFavorite": false,
          "name": "Chernobyl",
          "posterPath": "/hlLXt2tOPT6RRnjiUmoxyG1LTFi.jpg",
          "votesAverage": 8.6,
        }
      }
    />
    <TvShowCard
      key="60625"
      tvShow={
        Object {
          "backdropImagePath": "/mzzHr6g1yvZ05Mc7hNj3tUdy2bM.jpg",
          "firstAirDate": "2013-12-02",
          "id": 60625,
          "isFavorite": false,
          "name": "Rick and Morty",
          "posterPath": "/qJdfO3ahgAMf2rcmhoqngjBBZW1.jpg",
          "votesAverage": 8.6,
        }
      }
    />
  </div>
</section>
`;
